src/components/HistoryTable/HistoryTable.tsx

```diff
@@ -24,21 +24,42 @@ const HistoryTable: React.FC<HistoryTableProps> = ({
     isItemLoaded,
     toggleRowSizeAtIndex,
     commitCount,
+    selectedCommit,
+    markSelectedVisited,
+    processedCommitCount,
   } = useHistoryTable();
   const listRef = useRef<List>(null);
-
   useEffect(() => {
     if (recentlyFetchedCommits) {
       fetchNewCommit(recentlyFetchedCommits);
     }
     // Remove fetchNewCommit from the effect list to avoid infinite loop
   }, [recentlyFetchedCommits]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  // When we fetch new commits we need to tell react-window to re-render the list and update the heights for each of the rows since they will have changed based off of the new commits
   useEffect(() => {
-    if (listRef.current) {
-      listRef.current.resetAfterIndex(0);
+    if (processedCommitCount > 0) {
+      if (listRef.current) {
+        listRef.current.resetAfterIndex(0);
+      }
+    }
+  }, [processedCommitCount]);
+
+  useEffect(() => {
+    if (selectedCommit && selectedCommit.rowIndex && listRef.current) {
+      listRef.current.scrollToItem(selectedCommit.rowIndex, "center");
+      markSelectedVisited();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedCommit?.rowIndex]);
+
+  useEffect(() => {
+    if (selectedCommit) {
+      if (!selectedCommit.rowIndex) {
+        loadMoreItems();
+      }
     }
-  }, [recentlyFetchedCommits]);
+  }, [selectedCommit, loadMoreItems]);
```

Comment:
It almost seems like this hook could be combined with the one above it since the one above is basically `(selectedCommit && selectedCommit.rowIndex)` and the one here is `(selectedCommit && !selectedCommit.rowIndex)`. 🤔

Though maybe it would be more confusing. Just food for thought!!

Reply:
sgtm